src/core/gltf2importer/animationparser.cpp

```diff
@@ -277,6 +282,22 @@ Qt3DAnimation::QChannel AnimationParser::animationChannelDataFromBuffer(const QS
     if (nbComponents == 0 || valueStamps.empty())
         return channel;
 
+    // When dealing with morph targets we should have
+    // nbComponents * morphTargetCount * timeStamps.size() keyframe values
+    // nbComponents is 1 because we are dealing with a scalar weight type
+    // but the channel should actually expose one scalar weight for each morph targets
+    // -> one channel component for each morph target in the mesh
+    // 1 - n possibly
+    const bool isMorphTargetWeightChannel = (channelName == QStringLiteral("weights"));
```

Comment:
I'm a little confused by the last 3 lines of the comment, but that could be because I have no idea about morph targets.  Just check to make sure it makes sense.

Reply:
Right, I found it confusing it as well. Reworded it.